src/Compilers/CSharp/Portable/CSharpParseOptions.cs

```diff
@@ -50,90 +50,46 @@ public override IEnumerable<string> PreprocessorSymbolNames
             IEnumerable<string> preprocessorSymbols = null)
             : this(languageVersion, 
                   documentationMode, 
-                  kind, 
-                  preprocessorSymbols.ToImmutableArrayOrEmpty(), 
+                  kind,
+                  preprocessorSymbols.ToImmutableArrayOrEmpty(),
                   ImmutableDictionary<string, string>.Empty)
         {
-            // We test the mapped value, LanguageVersion, rather than the parameter, languageVersion,
-            // which has not had "Latest" mapped to the latest version yet.
-            if (!LanguageVersion.IsValid())
-            {
-                throw new ArgumentOutOfRangeException(nameof(languageVersion));
-            }
-
-            if (!kind.IsValid())
-            {
-                throw new ArgumentOutOfRangeException(nameof(kind));
-            }
-
-            if (preprocessorSymbols != null)
-            {
-                foreach (var preprocessorSymbol in preprocessorSymbols)
-                {
-                    if (!SyntaxFacts.IsValidIdentifier(preprocessorSymbol))
-                    {
-                        throw new ArgumentException($"{nameof(preprocessorSymbols)} contains a symbol that is not a valid identifier", nameof(preprocessorSymbols));
-                    }
-                }
-            }
         }
 
         internal CSharpParseOptions(
             LanguageVersion languageVersion,
             DocumentationMode documentationMode,
             SourceCodeKind kind,
             IEnumerable<string> preprocessorSymbols,
-            ImmutableDictionary<string, string> features)
-            : this(languageVersion, documentationMode, kind, preprocessorSymbols)
+            IReadOnlyDictionary<string, string> features)
+            : base(kind, documentationMode)
         {
-            if (features == null)
-            {
-                throw new ArgumentNullException(nameof(features));
-            }
-
-            _features = features;
+            this.SpecifiedLanguageVersion = languageVersion;
+            this.LanguageVersion = languageVersion.MapSpecifiedToEffectiveVersion();
+            this.PreprocessorSymbols = preprocessorSymbols.ToImmutableArrayOrEmpty();
+            _features = features?.ToImmutableDictionary() ?? ImmutableDictionary<string, string>.Empty;
```

Comment:
>features?.ToImmutableDictionary() [](start = 24, length = 33)

Is this going to allocate a new ImmutableDictionary even when parameter is already an ImmutableDictionary?